src/common/config_proxy.h

```diff
@@ -55,16 +66,20 @@ class ConfigProxy {
   }
   void diff(Formatter *f, const std::string& name=string{}) const {
     return config.diff(values, f, name);
+    Mutex::Locker l{lock};
```

Comment:
here should lock first then call config.diff, right?

Reply:
ahh, yeah! thanks.

Reply:
#23276